lib/rex/proto/kerberos/model/ticket_flags.rb

```diff
@@ -0,0 +1,74 @@
+# -*- coding: binary -*-
+
+module Rex
+  module Proto
+    module Kerberos
+      module Model
+        # TODO: Consolidate this with KdcOptionFlags, almost identical bar from the class defining the flags
+        # Represents the TicketFlags KerberosFlags.
+        # https://www.rfc-editor.org/rfc/rfc4120.txt - TicketFlags     ::= KerberosFlags
+        class TicketFlags
+          # @return [Integer] the integer value of the kerberos flags
+          attr_reader :value
+
+          # @param [Integer] value the numerical value of the flags
+          # @raise [ArgumentError] if any of the parameters are of an invalid type
+          def initialize(value)
+            raise ArgumentError, 'Invalid value' unless value.is_a?(Integer)
+
+            @value = value
+          end
+
+          # @param [Array<Integer,Rex::Proto::Kerberos::Model::TicketFlag>] flags an array of numerical values representing flags
+          # @return [Rex::Proto::Kerberos::Model::TicketFlags]
+          def self.from_flags(flags)
+            value = 0
+            flags.each do |flag|
+              value |= 1 << (31 - flag)
+            end
+
+            new(value)
+          end
+
+          def to_i
+            @value
+          end
+
+          # @param [Integer,Rex::Proto::Kerberos::Model::TicketFlag] flag the numerical value of the flag to test for.
+          # @return [Boolean] whether the flag is present within the current TicketFlags
+          def include?(flag)
+            ((value >> (31 - flag)) & 1) == 1
+          end
+
+          # @return [Boolean] whether the flag is present within the current TicketFlags
+          def enabled_flag_names
+            sorted_flag_names = TicketFlag.constants.sort_by { |name| TicketFlag.const_get(name) }
+            enabled_flag_names = sorted_flag_names.select { |flag| include?(TicketFlag.const_get(flag)) }
```

Comment:
It would be helpful to have a generic way to get the string name of a flag. Maybe something like this could be added to `TicketFlag`.

```ruby
      def self.name(value)
        constants.select { |c| c.upcase == c }.find { |c| const_get(c) == value }
      end
```